libraries/chain/scorum_evaluator.cpp

```diff
@@ -43,43 +46,13 @@ inline void validate_permlink_0_1(const string& permlink)
     {
         switch (c)
         {
-        case 'a':
-        case 'b':
-        case 'c':
-        case 'd':
-        case 'e':
-        case 'f':
-        case 'g':
-        case 'h':
-        case 'i':
-        case 'j':
-        case 'k':
-        case 'l':
-        case 'm':
-        case 'n':
-        case 'o':
-        case 'p':
-        case 'q':
-        case 'r':
-        case 's':
-        case 't':
-        case 'u':
-        case 'v':
-        case 'w':
-        case 'x':
-        case 'y':
-        case 'z':
-        case '0':
-        case '1':
-        case '2':
-        case '3':
-        case '4':
-        case '5':
-        case '6':
-        case '7':
-        case '8':
-        case '9':
+        // clang-format off
+        case 'a': case 'b': case 'c': case 'd': case 'e': case 'f': case 'g': case 'h': case 'i':
```

Comment:
change to
if (!std::islower(c) && !std::isdigit(c) && !(c == '-'))
             FC_ASSERT(false, "Invalid permlink character: ${s}", ("s", std::string() + c));